src/main/kotlin/defineDocumentation.kt

```diff
@@ -44,9 +47,50 @@ private fun writeAPI(w: PrintWriter, apiContainer: Container) {
 
   w.print("| ")
   val githubLink = apiContainer.url
-  val apidocsLink = apiContainer.properties.get("api-docs-url")
+  val apidocsLink = APIDocs.getFrom(apiContainer)
   w.print(" [GitHub](%s)".format(githubLink))
   w.print(" [APIdocs](%s)".format(apidocsLink))
 
   w.println("|")
 }
+
+@Suppress("DEPRECATION")
+private fun containersWithAPIDocs(model: Model): List<Container> {
+  val containers = model.softwareSystems
+    .sortedBy { it.name.toLowerCase() }
+    .flatMap { it.containers }
+
+  containers
+    .filter { it.url.orEmpty().contains("github.com/ministryofjustice") }
+    .forEach {
+      val oldUrl = APIDocs.getFrom(it)
+      val url = readAPIDocsURLFromRepoReadmeBadge(it)
+
+      if (url != null) {
+        if (oldUrl != null) {
+          println("[defineDocumentation] overriding API docs URL\n - from ${oldUrl}\n - to   $url")
+        }
+        APIDocs(url).addTo(it)
+      }
+    }
+
+  return containers.filter { APIDocs.getFrom(it) != null }
```

Comment:
Could we roll these functions into a single chain?

Reply:
done in 6f21e4af28371704702e0edca372bcdcc13702a1